src/XdebugHandler.php

```diff
@@ -260,7 +260,24 @@ private function doRestart($command)
         $this->tryEnableSignals();
         $this->notify(Status::RESTARTING, $command);
 
-        passthru($command, $exitCode);
+        // Use proc_open on linux/OSX and on recent Windows where stream_isatty may be used
+        // This is to ensure the process pipes remain bound to a tty which does not happen with passthru
+        if (function_exists('proc_open') && (!\defined('PHP_WINDOWS_VERSION_BUILD') || function_exists('stream_isatty'))) {
+            $proc = proc_open($command, array(), $pipes);
+            while ($status = @proc_get_status($proc)) {
+                if (!$status['running']) {
+                    break;
+                }
+                usleep(10000);
+            }
+            if (isset($status['exitcode'])) {
```

Comment:
Is there a reason why it would not be set?

Reply:
If $status did not get populated at all because proc_get_status failed somehow, or proc_open failed.

Reply:
proc_get_status will only fail if there is no resource, ie proc_open failed (which would require aborting the restart with an error message, I guess).

Reply:
or fall back on `passthru()` when `proc_open()` failed, that I think is the idea. Too shady? At most two times a fork fails. Maybe a little hang on Windows. @johnstevenson